sailthru/Sailthru_Client.php

```diff
@@ -1434,9 +1434,22 @@ protected function httpRequestCurl($action, array $data, $method = 'POST', $opti
         curl_setopt($ch, CURLOPT_HTTPHEADER, $this->httpHeaders);
         $response = curl_exec($ch);
         $this->lastResponseInfo = curl_getinfo($ch);
+
+        // Get the curl error message if there is one
+        $errorMessage = false === $response ? curl_error($ch) : null;
+
         curl_close($ch);
 
+        if (false === $response) {
+            // There's a curl error! throw an exception which gives us some details
+            throw new Sailthru_Client_Exception(
+                sprintf('Error with curl transport: %s', $errorMessage),
+                Sailthru_Client_Exception::CODE_TRANSPORT_ERROR
+            );
+        }
+
         if (!$response) {
+            // We have an empty (well, falsey) response for the server, but not a curl error
```

Comment:
I wasn't so sure about this one, then we'd need to throw the exception after the whole response was parsed into the headers/body.

I'm not sure that this "empty response" error has much real use, it's probably almost impossible for it to ever occur as we have asked curl for the headers. It would also be caught by the "Response: {$response} is not a valid JSON" exception, too.